unison-core/src/Unison/Name.hs

```diff
@@ -78,7 +104,10 @@ sortNamed' by by2 as = let
 
 unsafeFromText :: Text -> Name
 unsafeFromText t =
-  if Text.any (== '#') t then error $ "not a name: " <> show t else Name t
+  if useNewName then
+    undefined
+  else
+    if Text.any (== '#') t then error $ "not a name: " <> show t else Name t
```

Comment:
This should probably also support some syntax-specific escaping to allow '#' characters IMO.

Reply:
Yeah, agreed; by the time you are calling `unsafeFromText` you should know this is a valid `Name` per the syntax.